src/imitation/util/rollout.py

```diff
@@ -16,18 +16,44 @@
 
 from . import util  # Relative import needed to prevent cycle with __init__.py
 
-TrajectoryList = List[Dict[str, np.ndarray]]
-"""A list of trajectory dicts.
 
-Each dict contains the keys 'act', 'obs', and 'rew'. For details on these
-key-value pairs, see the docstring for `generate_trajectories`.
-"""
+class Trajectory(NamedTuple):
+  """A trajectory.
 
-TransitionsTuple = Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]
-"""An tuple of obs-act-obs-rew values.
+   Attributes:
+    obs: Observations, shape (trajectory_len+1, ) + observation_shape.
+    act: Actions, shape (trajectory_len, ) + action_shape.
+    rew: Reward, shape (trajectory_len, ).
+  """
+
+  act: np.ndarray
+  obs: np.ndarray
+  rew: np.ndarray
+
+
+class Transitions(NamedTuple):
+  """A batch of transition obs-act-obs-rew transitions.
+
+   Attributes:
+    old_obs: Old observations. Shape: (batch_size, ) + observation_shape.
+        The i'th observation `old_obs[i]` in this array is the observation seen
+        by the agent when choosing action `act[i]`.
+    act: Actions. Shape: (batch_size, ) + action_shape.
+    new_obs: New observation. Shape: (batch_size, ) + observation_shape.
+        The i'th observation `new_obs[i]` in this array is the observation after
+        the agent has taken action `act[i]`.
+    rew: Reward. Shape: (batch_size, ).
+        The reward `rew[i]` at the i'th timestep is received after the agent has
+        taken action `act[i]`.
+    done: Boolean array indicating episode termination. Shape: (batch_size, ).
+        `done[i]` is true iff `new_obs[i]` the last observatoin of an episode.
+  """
 
-For details see the docstring for `generate_transitions`.
-"""
+  old_obs: np.ndarray
```

Comment:
Most RL code I've seen uses `act` and `obs` to refer to observation seen at time `t` and action taken at time `t`, and `next_obs`, `nobs`, `obsn`, `obs_next`, etc. to refer to the next observation at time `t+1` (in analogy to the `(s,a,r,s')` convention in textbooks). The current naming scheme leads to confusing stuff like the `old_obs` vs. `act` that I highlighted in the BC code. IMO a better scheme would be `s/old_obs/obs/` and `s/new_obs/next_obs/` or something.